src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1386,12 +1378,32 @@ internal override ImmutableArray<Symbol> GetEarlyAttributeDecodingMembers(string
         {
             if (_lazyEarlyAttributeDecodingMembersDictionary == null)
             {
+                if (Volatile.Read(ref _lazyMembersDictionary) is Dictionary<string, ImmutableArray<Symbol>> result)
+                {
+                    return result;
+                }
+
                 var membersAndInitializers = GetMembersAndInitializers(); //NOTE: separately cached
 
                 // NOTE: members were added in a single pass over the syntax, so they're already
                 // in lexical order.
 
-                var membersByName = membersAndInitializers.NonTypeNonIndexerMembers.ToDictionary(s => s.Name);
+                Dictionary<string, ImmutableArray<Symbol>> membersByName;
+
+                if (!membersAndInitializers.HaveIndexers)
+                {
+                    membersByName = membersAndInitializers.NonTypeMembers.ToDictionary(s => s.Name);
```

Comment:
>s => s.Name [](start = 87, length = 11)

Nit: consider adding `static` to these lambdas (and below).

Reply:
> Nit: consider adding static to these lambdas (and below).

I see no benefit in doing so for simple lambdas like that.

---
In reply to: [568848641](https://github.com/dotnet/roslyn/pull/50961#discussion_r568848641) [](ancestors = 568848641)